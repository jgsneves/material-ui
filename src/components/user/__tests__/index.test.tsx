import { render } from '@testing-library/react';
import User from '..';

describe('User component test suite', () => {
  it('should render letter avatar', () => {
    const { asFragment } = render(
      <User
        name="José da Silva"
        email="jose@teste.com.br"
        mobilePhone="51 99145221"
      />,
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it('should render image avatar', () => {
    const { asFragment } = render(
      <User
        name="José da Silva"
        email="jose@teste.com.br"
        mobilePhone="51 99145221"
        photoUrl="photo.png"
      />,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
