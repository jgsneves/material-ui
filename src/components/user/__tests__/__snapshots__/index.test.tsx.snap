// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User component test suite should render image avatar 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1syf4pq-MuiPaper-root-MuiCard-root"
  >
    <div
      class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
    >
      <div
        class="MuiAvatar-root MuiAvatar-circular css-a2xf8u-MuiAvatar-root"
      >
        <img
          alt="José da Silva"
          class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
          src="photo.png"
        />
      </div>
      <div
        class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-h93ljk-MuiTypography-root"
      >
        José da Silva
      </div>
      <p
        class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
      >
        jose@teste.com.br
      </p>
      <p
        class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
      >
        51 99145221
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`User component test suite should render letter avatar 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1syf4pq-MuiPaper-root-MuiCard-root"
  >
    <div
      class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
    >
      <div
        class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-8h65ya-MuiAvatar-root"
      >
        J
      </div>
      <div
        class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-h93ljk-MuiTypography-root"
      >
        José da Silva
      </div>
      <p
        class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
      >
        jose@teste.com.br
      </p>
      <p
        class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
      >
        51 99145221
      </p>
    </div>
  </div>
</DocumentFragment>
`;
